### ISSUE TITLE
html tables rendered from markdown don't look right with latest pandoc (3.3) 

### ISSUE THREAD

**Issue by therealgenna** [rstudio/rmarkdown#2567]
I am running `rmarkdown` within `R`. In the end it's calling `pandoc` to create html output. The command printed on the console is like that:
```
PANDOC=<path>/env/bin/pandoc
$PANDOC +RTS -K512m -RTS \
  run.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash \
  --output run.html \
  --lua-filter $ENV/lib/R/library/rmarkdown/rmarkdown/lua/pagebreak.lua \
  --lua-filter $ENV/lib/R/library/rmarkdown/rmarkdown/lua/latex-div.lua \
  --embed-resources --standalone --variable bs3=TRUE --section-divs \
  --table-of-contents --toc-depth 3 --variable toc_float=1 --variable toc_selectors=h1,h2,h3 --variable toc_collapsed=1 \
  --variable toc_smooth_scroll=1 --variable toc_print=1 \
  --template $ENV/lib/R/library/rmarkdown/rmd/h/default.html \
  --no-highlight --variable highlightjs=1 --variable theme=bootstrap \
  --css leftjustify.css \
  --mathjax --variable 'mathjax-url=https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' \
  --include-in-header /tmp/RtmpnvLnyX/rmarkdown-str16204b65985cf3.html \
  --variable code_folding=hide --variable code_menu=1
```
I have two environments, an old one with pandoc 3.1.3 and a new one with 3.3. Both environments produce I believe a valid `.md` file with a table that looks like
```
Table: Original cell counts

|Sample | Freq|
|:------|----:|
|Leah1  | 2938|
|Leah3  | 5847|
|Leah4  | 5307|
|Leah6  | 5757|
|Leah7  | 5897|
|Leah8  | 5536|
|T1     | 8703|
|T2     | 9052|
```
However, the new pandoc (3.3) makes ugly html tables, e.g.:
![image](https://github.com/user-attachments/assets/39b4d37c-2f0c-4ec7-8913-6b14869e6a56)

While the older one is what I expect:
![image](https://github.com/user-attachments/assets/44a9627b-1203-4c2a-9f39-3f08676c7efc)

I've tracked the problem to the change in pandoc 3.2.1, I believe.
It got rid of header, odd, even classes on tr (https://github.com/jgm/pandoc/issues/9376) in version 3.2.1 (linked from https://pandoc.org/releases.html). If I add these "classes" manually back to the final html, it then looks fine.

Here on the left is new version, without "classes", and on the right it's my manual edits, adding "classes":
```
<table>                                                         <table>
<caption>Original cell counts</caption>                         <caption>Original cell counts</caption>
<thead>                                                         <thead>
<tr>                                                          | <tr class="header">
<th align="left">Sample</th>                                    <th align="left">Sample</th>
<th align="right">Freq</th>                                     <th align="right">Freq</th>
</tr>                                                           </tr>
</thead>                                                        </thead>
<tbody>                                                         <tbody>
<tr>                                                          | <tr class="odd">
<td align="left">Leah1</td>                                     <td align="left">Leah1</td>
<td align="right">2938</td>                                     <td align="right">2938</td>
</tr>                                                           </tr>
<tr>                                                          | <tr class="even">
<td align="left">Leah3</td>                                     <td align="left">Leah3</td>
<td align="right">5847</td>                                     <td align="right">5847</td>
</tr>                                                           </tr>
<tr>                                                          | <tr class="odd">
<td align="left">Leah4</td>                                     <td align="left">Leah4</td>
<td align="right">5307</td>                                     <td align="right">5307</td>
</tr>                                                           </tr>
...                                                             ...
```
Adding the classes back, as on the right, fixes the issue and the table looks good in the web browsers.

**I've first submitted this issue to `pandoc` developers (https://github.com/jgm/pandoc/issues/10073) however they said that I "may need to consult with RMarkdown maintainers, who will know more about how the CSS is applied; they will need to update things, probably, to work with the new pandoc defaults.". This is what I'm doing now.**

Also, it might be that this is related to https://github.com/rstudio/rmarkdown/issues/2566

Pandoc version 3.3 (on Linux)
rmarkdown 2.27
knitr 1.48

**Comment by yihui** [comment]
Should be fixed now. Thanks for the report! You can test the development version via

```r
remotes::install_github('rstudio/rmarkdown')
```

**Comment by franceskoi** [comment]
Thanks for the post. My Pandoc version is 3.2.1.

**Comment by yihui** [comment]
@franceskoi The fix applies to Pandoc >= 3.2.1, so you need to install the dev version mentioned above. Thanks!
